src/tribler/core/components/libtorrent/restapi/tests/test_downloads_endpoint.py

```diff
@@ -153,8 +153,9 @@ async def test_get_downloads_no_downloads(mock_dlmgr, rest_api):
     """
     Testing whether the API returns an empty list when downloads are fetched but no downloads are active
     """
+    expected_json = {"downloads": [], 'checkpoints': {'total': 10, 'loaded': 10, 'all_loaded': True}}
     result = await do_request(rest_api, 'downloads?get_peers=1&get_pieces=1',
-                              expected_code=200, expected_json={"downloads": []})
+                              expected_code=200, expected_json=expected_json)
```

Comment:
Probably we could add a test for intermediate state of download loading (like "a half loaded")

Reply:
You are right, I added checks for half-loaded state as well